File: 367-valid-perfect-square/valid-perfect-square.c
bool isPerfectSquare(int num) {
    long int start = 1;
    long int end = num;
    while (start<=end){
        long int mid = (start+end)/2;
        if (mid*mid == num)
            return true;
        else if (mid*mid < num)
            start =mid+1;
        else
            end=mid-1;
    }
    return false;
}

File: 367-valid-perfect-square/valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:    
        start = 1
        end = num
        while start <= end:
            mid = (start+end) // 2
            p = mid * mid
            if p == num:
                return True
            if p < num:
                start=mid+1
            else:
                end= mid-1
        return False
        